Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -624,13 +622,11 @@ public class HTTPClient {
                 proxy: proxy,
                 ignoreUncleanSSLShutdown: ignoreUncleanSSLShutdown,
                 decompression: decompression,
-                // TODO: set to automatic by default
-                httpVersion: .http1Only
+                httpVersion: .automatic
             )
         }
 
-        // TODO: make public
-        internal init(
+        public init(
```

Comment:
TBH: I don't think we should add yet another initializer for HTTPClient.Configuration. I think we should deprecate all of them, except for `init()`. In `init()` we set all default values and then the user can change the config, since it is just a struct. We have precedent for this move in NIOSSL.

Reply:
If I understand you correct, a user will then only be able to change the config by create a temporary variable, initialising it with our default config, mutating it and passing it to the `HTTPClient` initialiser, right? This would be a rather big change because we would need to update all our tests if we deprecate all other inits. I would rather do that in another PR if we would like to do that before merging this one.

Reply:
cc @Lukasa

Reply:
We could also think about adding an extension to `HTTPClient.Configuration` so we can change the configuration through daisy chaining like [I have added in RSocket](https://github.com/rsocket/rsocket-swift/blob/df80e64fa0d16e00b1a192c475e47b87c919c0a1/Sources/RSocketCore/Client/ClientConfiguration.swift#L107-L115):
```swift
extension HTTPClient.Configuration {
    /// Creates a copy and updates the property described by `keyPath` with the given `newValue`
    /// - Returns: updated configuration
    public func set<Value>(_ keyPath: WritableKeyPath<Self, Value>, to newValue: Value) -> Self {
        var copy = self
        copy[keyPath: keyPath] = newValue
        return copy
    }
}
```

This would easy the transition because you probably end up with the same amount of lines as a config and a user does not have to add a temporary variable.

Reply:
When we are already on it, what about nested types `HTTPClient.Configuration` uses e.g. `HTTPClient.Configuration.ConnectionPool` and `HTTPClient.Configuration.Timeout`? They will probably suffer from the same problem, but on a much smaller scale.

Reply:
I'm inclined to tackle this in a separate PR. For now we don't need to add a new init for this feature, we can just exclude it from the pre-existing inits.